homeassistant/components/zha/helpers.py

```diff
@@ -419,11 +419,22 @@ def zha_device_info(self) -> dict[str, Any]:
     @callback
     def handle_zha_event(self, zha_event: ZHAEvent) -> None:
         """Handle a ZHA event."""
+        if ATTR_UNIQUE_ID in zha_event.data:
+            unique_id = zha_event.data.pop(ATTR_UNIQUE_ID)
```

Comment:
is it okay to mutate `zha_event` ? Altnerative approach is to specify `**zha_event.data` before the other keys, so manually set keys always win.

Reply:
Fixed. I'm going to play it safe and keep the same order, putting `ATTR_UNIQUE_ID` last.